regression/cbmc/dynamic_struct/main.c

```diff
@@ -0,0 +1,55 @@
+#include <assert.h>
+#include <stdlib.h>
+
+#ifndef NULL
+#  define NULL ((void *)0)
+#endif
+
+struct list;
+
+typedef struct list list_nodet;
+
+list_nodet fill_node(signed int depth_tag_list);
+
+struct list
+{
+  int datum;
+  struct list *next;
+};
+
+int max_depth = 4;
+
+list_nodet *build_node(int depth)
+{
+  if(max_depth < depth)
+    return ((list_nodet *)NULL);
+  else
+  {
+    list_nodet *result = malloc(16);
```

Comment:
this should be changed to `sizeof(struct list)` so it doesn't randomly break in the future

Reply:
Agreed that this should be changed to something so that at least the value is self-documented.

Reply:
The bug only appears if I don't use `sizeof`.

Reply:
Because `malloc(sizeof(type))` is special-cased to produce a `type*` instead of a `char*`